includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -1112,7 +1112,7 @@ public function find_matching_product_variation( $product, $match_attributes = a
 
 			// Loop over the variation meta keys and values i.e. what is saved to the products. Note: $attribute_value is empty when 'any' is in use.
 			foreach ( $variation as $attribute_key => $attribute_value ) {
-				$match_any_value = empty( $attribute_value );
+				$match_any_value = empty( $attribute_value ) && 0 != $attribute_value;
```

Comment:
not a blocker, but I think we should use `!==` here.

Reply:
I actually purposefully didn't use strict comparison in order to match also '0.0' or 0 or 0.0

Maybe it would be better to be explicit, though.

Reply:
Maybe we should check for an empty string instead of using empty? I think we might get the same results if you use false or 0.0 since empty will return true on all of those. All we want to check here is if the value is an empty string?

Reply:
I agree with @kloon 's suggestion above. if we check for the empty string then I guess we won't need to do any other comparison right? wdyt @peterfabian ?

Reply:
I was trying to be cautious in case the value could be NULL or something else that `empty` would evaluate to true. 
Fixed to only check empty string now.